rotkehlchen/chain/evm/decoding/paraswap/decoder.py

```diff
@@ -98,33 +99,42 @@ def _decode_swap(
             log.error(f'Could not find the corresponding events when decoding paraswap swap {context.transaction.tx_hash.hex()}')  # noqa: E501
             return DEFAULT_DECODING_OUTPUT
 
-        if partial_refund_event is not None:  # if some in_asset is returned back
-            if in_event.asset == out_event.asset:  # check the similarity of asset with out_event
-                # due to order of events it can happen that this two get mislabeled and need to be swapped  # noqa: E501
-                partial_refund_event, in_event = in_event, partial_refund_event
+        if partial_refund_event is not None:  # if some in_asset is returned back.
+            # it's assumed in the above for loop, that the second in_event is the partial refund
+            # we verify below if the assumption is wrong, by checking the asset's similarity
+            if in_event.asset == out_event.asset:  # if this is true, then assumption is wrong
+                partial_refund_event, in_event = in_event, partial_refund_event  # swap them
+            # we check it here instead of in the for loop above because, out_event could be None
+
             out_event.balance.amount -= partial_refund_event.balance.amount  # adjust the amount
             context.decoded_events.remove(partial_refund_event)  # and remove it from the list
         out_event.notes = f'Swap {out_event.balance.amount} {out_event.asset.resolve_to_asset_with_symbol().symbol} in paraswap'  # noqa: E501
 
-        fee_raw = fee_asset = None  # extract the fee info
-        if in_event.asset.is_evm_token():
-            fee_asset = in_event.asset.resolve_to_evm_token()
-            for log_event in context.all_logs:
-                if (
-                    log_event.topics[0] == ERC20_OR_ERC721_TRANSFER and
-                    hex_or_bytes_to_address(log_event.topics[1]) == PARASWAP_AUGUSTUS_ROUTER and
-                    hex_or_bytes_to_address(log_event.topics[2]) == self.fee_receiver_address and
-                    log_event.address == fee_asset.evm_address
-                ):
-                    fee_raw = hex_or_bytes_to_int(log_event.data)
-                    break
+        in_asset = in_event.asset.resolve_to_evm_token() if in_event.asset.is_evm_token() else in_event.asset  # noqa: E501
+        out_asset = out_event.asset.resolve_to_evm_token() if out_event.asset.is_evm_token() else out_event.asset  # noqa: E501
```

Comment:
instead of resolving to evm don't you need to resolve to crypto asset? evmToken is a subclass

```suggestion
        in_asset = in_event.asset.resolve_to_crypto_asset()
        out_asset = out_event.asset.resolve_to_crypto_asset()
```